code/messaging/build.gradle

```diff
@@ -23,13 +23,17 @@ android {
         targetSdkVersion rootProject.ext.targetSdkVersion
         versionCode 1
         versionName rootProject.moduleVersion
+        // flag for e2e tests
+        buildConfigField "java.util.concurrent.atomic.AtomicBoolean", "IS_E2E_TEST", "new java.util.concurrent.atomic.AtomicBoolean(false)"
+        // flag for functional tests
+        buildConfigField "java.util.concurrent.atomic.AtomicBoolean", "IS_FUNCTIONAL_TEST", "new java.util.concurrent.atomic.AtomicBoolean(false)"
 
         testInstrumentationRunner rootProject.ext.testInstrumentationRunner
         consumerProguardFiles 'consumer-rules.pro'
     }
 
     flavorDimensions "target"
-    productFlavors{
+    productFlavors {
         phone {
             dimension "target"
```

Comment:
nit: Since we have only one flavorDimension we can refrain from mentioning explicitly in the product flavors.
Android gradle plugin will do it implicitly for us.